src/formatter.ts

```diff
@@ -575,19 +684,20 @@ export class Formatter {
 
         return { expectedDistance: 0, fromTickablePx: 0, maxNegativeShiftPx: 0 };
       });
+      return distances;
     }
 
-    function shiftToIdealDistances(idealDistances) {
+    function shiftToIdealDistances(idealDistances: Distance[]): number {
       // Distribute ticks to the contexts based on the calculated distance error.
       const centerX = adjustedJustifyWidth / 2;
       let spaceAccum = 0;
 
       contextList.forEach((tick, index) => {
         const context = contextMap[tick];
         if (index > 0) {
-          const x = context.getX();
+          const contextX = context.getX();
           const ideal = idealDistances[index];
-          const errorPx = ideal.fromTickable.getX() + ideal.expectedDistance - (x + spaceAccum);
+          const errorPx = (ideal.fromTickable?.getX() ?? 0) + ideal.expectedDistance - (contextX + spaceAccum);
```

Comment:
This is a silent failure, we should use an assertion instead.

Reply:
done

Reply:
done